File: ios/core_old/listeners/AwesomeActionEventListener.swift
//
//  AwesomeActionEventListener.swift
//  awesome_notifications
//
//  Created by CardaDev on 30/01/22.
//

import Foundation

public protocol AwesomeActionEventListener: AnyObject {
    func onNewActionReceived(fromEventNamed eventName:String, withActionReceived actionReceived:ActionReceived)
    func onNewActionReceivedWithInterruption(fromEventNamed eventName:String, withActionReceived actionReceived:ActionReceived) -> Bool
}

File: example/android/awn_core/src/main/java/me/carda/awesome_notifications/core/listeners/AwesomeActionEventListener.java
package me.carda.awesome_notifications.core.listeners;

import me.carda.awesome_notifications.core.models.returnedData.ActionReceived;

public interface AwesomeActionEventListener {
    public void onNewActionReceived(String eventName, ActionReceived actionReceived);
    public boolean onNewActionReceivedWithInterruption(String eventName, ActionReceived actionReceived);
}
